File: go-sdk/pkg/models/operations/getfolderlocks.go
package operations

import (
	"openapi/pkg/models/shared"
	"openapi/pkg/utils"
)

type GetFolderLocksQueryParams struct {
	FolderID string `queryParam:"style=form,explode=true,name=folder_id"`
}

type GetFolderLocksRequest struct {
	Retries     *utils.RetryConfig
	QueryParams GetFolderLocksQueryParams
}

type GetFolderLocksResponse struct {
	ClientError *shared.ClientError
	ContentType string
	FolderLocks *shared.FolderLocks
	StatusCode  int64
}

File: sdks/ts/src/sdk/models/operations/getfolderlocks.ts
import { SpeakeasyMetadata, SpeakeasyBase, RetryConfig } from "../../../internal/utils";
import * as shared from "../shared";



export class GetFolderLocksQueryParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=folder_id" })
  folderId: string;
}


export class GetFolderLocksRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  retries?: RetryConfig;

  @SpeakeasyMetadata()
  queryParams: GetFolderLocksQueryParams;
}


export class GetFolderLocksResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  clientError?: shared.ClientError;

  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  folderLocks?: shared.FolderLocks;

  @SpeakeasyMetadata()
  statusCode: number;
}
